app/controller/AppController.php

```diff
@@ -65,25 +64,41 @@ public function accumulate(string $value)
     {
         try {
             $this->manager->append($value);
-            $chaine = $this->manager->getAccumulator();
-            $this->index($chaine);
-        } catch(Exception $e) {
+            $this->index();
+        } catch (Exception $e) {
             $this->error($e->getMessage());
         }
     }
 
     /**
      * Lance une action (opération, égal, pourcentage, ...)
      * @param string $action
+     * @throws Exception
      */
     public function action(string $action)
     {
-        switch($action) {
-            case 'clear';
+        switch ($action) {
+            case 'clear':
                 $this->manager->reset();
                 break;
-            default;
+            case 'plus':
+            case 'minus':
+            case 'times':
+            case 'divide':
+                $this->manager->operator($action);
+                break;
+            case 'equals':
+                try {
+                    $this->manager->calculate();
+                } catch (Exception $e) {
+                    $this->error($e->getMessage());
+                    die;
```

Comment:
A mon avis ce `die()` n'est pas nécessaire si ?

Reply:
Si je ne met pas le die, il exécute le $this->index() malgré le catch, ce que j en comprends pas.
Je croyais que le catch arrêtais le code puis qu'il y a l'erreur.